lib/ClangImporter/ImportDecl.cpp

```diff
@@ -9022,11 +9028,18 @@ Decl *ClangImporter::Implementation::importDeclAndCacheImpl(
 
   auto Canon = cast<clang::NamedDecl>(UseCanonicalDecl? ClangDecl->getCanonicalDecl(): ClangDecl);
 
-  if (auto Known = importDeclCached(Canon, version, UseCanonicalDecl)) {
-    if (!SuperfluousTypedefsAreTransparent &&
-        SuperfluousTypedefs.count(Canon))
+  Optional<Decl *> optionalKnown =
+      importDeclCached(Canon, version, UseCanonicalDecl);
+  if (optionalKnown.hasValue()) {
```

Comment:
Nit: a cool trick is you can write the following:
```
if (Optional<Decl *> optionalKnown = importDeclCached(Canon, version, UseCanonicalDecl)) { ...
```

Reply:
Done.